### PR TITLE
CKS-334 Scaffold chapter pages

### PR DESCRIPTION
**Pull request by ediblecode**
https://nicedigital.atlassian.net/browse/CKS-334

### CODE REVIEW

File: gatsby/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "version": "1.0.0",
   "description": "Website for NICE Clinical Knowledge Summaries (CKS) built with Gatsby and using the NICE Design System",
   "scripts": {
+    "preinstall": "npx npm-force-resolutions",
```
[the remark this answers is not in the document]
Comment (ediblecode) [reply gatsby/package.json:7]: Shouldn't do, as long as you've installed Node and npm then this should just work fine.

File: gatsby/src/components/ChapterBody/ChapterBody.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		headingLevel == 2 ? ` class="visually-hidden">` : `>`,
+		`${stripHtmlTags(chapter.htmlHeader)}`,
+		`</h${headingLevel}>`,
+	].join("");
+
+	return (
+		<section aria-labelledby={chapter.slug}>
+			<div
+				className={styles.section}
+				dangerouslySetInnerHTML={{
```
[the remark this answers is not in the document]
Comment (ediblecode) [reply gatsby/src/components/ChapterBody/ChapterBody.tsx:44]: Interesing. I didn't. I'm not a fan of the extra ref as it seems to add complication. I think in our case we're not rendering loads of DOM nodes so the performance impact would be negligible. So would rather leave as-is

[the remark this answers is not in the document]
Comment (ediblecode) [reply gatsby/src/components/ChapterBody/ChapterBody.tsx:44]: Good find though!

File: gatsby/src/templates/ChapterLevel1/ChapterLevel1.test.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+		it("should render custom nav list with aria label", () => {
+			const { getByLabelText } = render(
+				<ChapterLevel1Page
+					{...({ pageContext: { chapter } } as ChapterLevel1PageProps)}
+				/>
+			);
+			expect(
+				getByLabelText("Pages within Background information")
+			).toHaveProperty("tagName", "UL");
```
[the remark this answers is not in the document]
Comment (ediblecode) [reply gatsby/src/templates/ChapterLevel1/ChapterLevel1.test.tsx:148]: We do as we're using `getByLabelText` which will use the aria label

File: gatsby/src/templates/ChapterLevel1/ChapterLevel1.test.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	describe("no html string content/custom nav", () => {
+		beforeEach(() => {
+			chapter.htmlStringContent = "<!-- No content -->";
+		});
+
+		it("should render visually hidden heading 2 with html header content", () => {
+			const { getByText } = render(
+				<ChapterLevel1Page
+					{...({ pageContext: { chapter } } as ChapterLevel1PageProps)}
+				/>
```
[the remark this answers is not in the document]
Comment (ediblecode) [reply gatsby/src/templates/ChapterLevel1/ChapterLevel1.test.tsx]: Good shout, I've refactored that.

File: gatsby/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (1996dylanriley) [reply gatsby/package.json:7]: appologies for my typo it should read "Is this going to cause issues for those who havent set up npx when they run npm i?"

[the remark this answers is not in the document]
Comment (ediblecode) [reply gatsby/package.json:7]: npx is installed as part of npm so should be fine

[the remark this answers is not in the document]
Comment (1996dylanriley) [reply gatsby/package.json:7]: just ran npm i and confirmed thats it works thanks. Will apporve

**Pull request merged**